### ISSUE TITLE
Modal: Autofocus not working in some cases

### ISSUE THREAD

**Issue by nbcp** [infor-design/enterprise#4409]
**Describe the bug**
Some variants of modal and messages does not autoFocus. `autoFocus=true` is the default setting.
There are two scenarios where I observed this happens.
1. The modal contains a link - the focus highlight does not show
2. The modal only has text content and 1 button (usually cancel) which is not set as default.

**To Reproduce**
1. Go to https://design.infor.com/code/ids-enterprise/latest/demo/components/message/example-index
2. Click on "Success Example" or "Allow Tags"

**Expected behavior**
Link should be focused.

Also if modal content only contains text and there is no default button (when we don't want a blue button), it should focus the 1st button. The focusable element loses because of this part in the code:
````
        if (focusElem.length === 1 && focusElem.is('.btn-modal')) {
          focusElem = thisElem.element.find('.btn-modal-primary');
        } // If the selected element is a tab, actually make sure it's the "selected" tab.
````

**Version**

- ids-enterprise: v4.32

**Screenshots**

An example of our use case where there is only text and only the modal buttons can be focused but we don't want either to be default (blue color) 
![image](https://user-images.githubusercontent.com/11013464/93976164-2201ce00-fdab-11ea-9762-d31646eb1dcf.png)

**Additional context**
These scenarios above are issues reported from VPAT

**Comment by tmcconechy** [comment]
The original functionality on autoFocus was inputs. So i can see why these cases probably don't work. Can try and make it focus the modal button / link instead if thats first.

**Comment by nbcp** [comment]
Hi @tmcconechy any update on this one?

**Comment by tmcconechy** [comment]
I put it in for next sprint.

**Comment by janahintal** [comment]
This issue is now resolved. Thanks. 
Verified in https://master-enterprise.demo.design.infor.com/components/message/test-focus-anchor-and-no-default-button.html?theme=uplift&variant=light&colors=0066D4
